### PR TITLE
models: clean up

### PR DESCRIPTION
**Pull request by lnielsen**
* Renames Object to ObjectVersion to ensure clear naming of concepts.

* Adds bucket snapshotting feature and tests.

* Adds ObjectVersion copy and restore methods.

* Adds documentation to models.

Signed-off-by: Lars Holm Nielsen <lars.holm.nielsen@cern.ch>

### CODE REVIEW

File: invenio_files_rest/models.py [hunk 1]
```diff
@@ -316,24 +453,105 @@ def set_contents(self, stream, size=None, chunk_size=None):
...
+        """Send file to client.
+
+        :returns: Response.
+        """
         return self.file.send_file(self)
 
+    def restore(self):
+        """Restore version of an object.
+
+        Raises an exception if the object.
```
Comment (egabancho) [inline invenio_files_rest/models.py]: I guess you forgot something here :wink:

File: invenio_files_rest/storage.py [hunk 2]
```diff
@@ -60,26 +60,34 @@ def make_path(self):
         )
```
Comment (egabancho) [inline invenio_files_rest/storage.py]: Copyright

File: docs/index.rst [hunk 3]
```diff
@@ -44,9 +44,9 @@ If you are looking for information on a specific function, class or method,
 this part of the documentation is for you.
```
Comment (egabancho) [inline docs/index.rst]: Copyright

**Comment by lnielsen** [comment]
Ready to go

**Pull request merged**